appsec/src/extension/commands/request_shutdown.c

```diff
@@ -189,3 +261,355 @@ static void _pack_headers_no_cookies_map(
     ZEND_HASH_FOREACH_END();
     mpack_finish_map(w);
 }
+
+static const char *nullable _header_content_type_zend_array(
+    const zend_array *nonnull hl, size_t *nonnull len)
+{
+
+    zend_string *key;
+    zval *val;
+    zend_long idx;
+    ZEND_HASH_FOREACH_KEY_VAL((zend_array *)hl, idx, key, val)
+    {
+        (void)idx;
+        if (!key) {
+            continue;
+        }
+        if (zend_string_equals_literal_ci(key, "content-type")) {
+            if (Z_TYPE_P(val) == IS_STRING) {
+                *len = Z_STRLEN_P(val);
+                return Z_STRVAL_P(val);
+            }
+            if (Z_TYPE_P(val) == IS_ARRAY) {
+                zend_array *arr = Z_ARR_P(val);
+                HashPosition pos;
+                zend_hash_internal_pointer_end_ex(arr, &pos);
+                zval *zv = zend_hash_get_current_data_ex(arr, &pos);
+                if (!zv) {
+                    continue;
+                }
+                ZVAL_DEREF(zv);
+                if (Z_TYPE_P(zv) != IS_STRING) {
+                    continue;
+                }
+                *len = Z_STRLEN_P(zv);
+                return Z_STRVAL_P(zv);
+            }
+        }
+    }
+    ZEND_HASH_FOREACH_END();
+
+    return NULL;
+}
+
+static zval _convert_json(char *nonnull entity, size_t entity_len)
+{
+    zval zv;
+    ZVAL_NULL(&zv);
+#define MAX_DEPTH 12
```

Comment:
We did talk about using the same limits as the WAF so it might be good to start thinking about that as well...

Reply:
unfortunately, it seems that PHP's json_decode just gives up and returns nothing if the max depth is exceeded. So this should be fairly high.